### PR TITLE
TDL-18578 : Errors Observed in Tap Failure Dashboard

### PR DESCRIPTION
**Pull request by shantanu73**
# Description of change
(write a short description or paste a link to JIRA)
1) Added a new schema chat_webpath_array.json for chat webpath which validates array elements.
2) Modified chats.json to include chat_webpath_array.json schema for webpath in chats API.

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_zendesk_chat/schemas/chats.json [hunk 1]
```diff
@@ -41,7 +41,14 @@
     },
     "webpath": {
       "items": {
-        "$ref": "chat_webpath"
+        "anyOf": [
+          {
+            "$ref": "chat_webpath"
+          },
+          {
+            "$ref": "chat_webpath_array"
```
Comment (dbshah1212) [inline tap_zendesk_chat/schemas/chats.json:49]: Instead of creating 2 separate files, we can also put anyOf inside the chat_webpath file.

**Review by nitingaikwad1 — APPROVED** [review]

File: tap_zendesk_chat/schemas/chats.json (hunk 1 above)
Comment (shantanu73) [reply tap_zendesk_chat/schemas/chats.json:49]: Okay, will make this change.

File: tap_zendesk_chat/schemas/chat_webpath_array.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+  "items": {
+    "type": [
+      "null",
+      "string",
+      "integer"
```
[the remark this answers is not in the document]
Comment (shantanu73) [reply tap_zendesk_chat/schemas/chat_webpath_array.json:6]: The schema is able to validate the input json irrespective of the order in which these data types are kept. 
The integration tests ran fine and I've validated it through online schema validator as well.
So, wanted to confirm if you're suggesting it as a best practice or does it actually impact any given input ?

[the remark this answers is not in the document]
Comment (luandy64) [reply tap_zendesk_chat/schemas/chat_webpath_array.json]: The issue is that the `singer.Transformer` cares about the order.

Here's an example

```python
$ python3
>>> from singer import Transformer
>>> schema = {
...   "type": "object",
...   "properties": {
...     "string-int": {"type": ["null", "string", "integer"]},
...     "int-string": {"type": ["null", "integer", "string"]}
...   }
... }
>>> records = [
...   {"string-int": 1, "int-string": 1},

...   {"string-int": "2", "int-string": "2"}
... ]
>>> with Transformer() as tm:
...   for record in records:
...     print(tm.transform(record, schema))
...
{'string-int': '1', 'int-string': 1}
{'string-int': '2', 'int-string': 2}
```

Notice that
1. `string-int` is a field that prioritizes `string`
2. `int-string` is a field that prioritizes `integer` 
3. The transformer forces `string-int` to be output as a string
4. The transformer forces `int-string` to be output as an integer

**Review by luandy64 — APPROVED** [review]

**Pull request merged**